subsys/bluetooth/mesh/main.c

```diff
@@ -163,18 +157,20 @@ int bt_mesh_provision(const uint8_t net_key[16], uint16_t net_idx,
 		bt_mesh_net_store();
 	}
 
-	bt_mesh_start();
+	atomic_set_bit(bt_mesh.flags, BT_MESH_VALID);
+
+	return bt_mesh_start();
```

Comment:
Why flag is not reset if `bt_mesh_start` failed?

Reply:
Validity of settings is not related to start. Mesh can be started at any time if it can't start now for some reason.